openedx/openedx-platform
# feat: deprecate `track_function` and `publish` in ModuleSystem [BD-13]

## Description

As a part of the ModuleSystem removal, we want to get rid of its remaining attributes:
- `track_function`,
- `publish`.

## Supporting information

Sandbox instance: https://pr30046.sandbox.opencraft.hosting/
Use `staff@example.com:edx` to log in.

## Testing instructions
1. We have replaced the remaining usages of `track_function` with `publish`, so it makes sense to check these two together. 
1. Log into [the sandbox instance](https://manage.opencraft.com/instance/33870/edx-appserver/24796/) (private link) and check tracking logs for the next actions.
1. Check that the grading is working. You can use [this XBlock](https://app.pr30046.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@basic_questions/block-v1:edX+DemoX+Demo_Course+type@vertical+block@2152d4a4aadc4cb0af5256394a3d1fc7) in the LMS and Preview. Check that the [Studio view](https://studio.pr30046.sandbox.opencraft.hosting/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@2152d4a4aadc4cb0af5256394a3d1fc7) is working correctly too.
1. Check that the "Show answer" button emits events.
1. Check that the completion events are working (the completion is already enabled on the sandbox).

## Other information

Private-ref: [BB-5518](https://tasks.opencraft.com/browse/BB-5518)
